store/AuthContext.js

```diff
@@ -45,18 +45,29 @@ export function AuthProvider({ children }) {
           emailVerified: user.emailVerified,
           accountType: user.photoURL,
         }
-        const token = await user.getIdToken(true)
-
+        const token = await user.getIdToken()
         // fetchUserInfo(userObject)
         setCurrentUser(userObject)
-        nookies.set(undefined, 'token', token, {})
+        nookies.set(undefined, 'token', token, { path: '/' })
       }
       setIsLoading(false)
     })
     // onAuthStateChanged accepts a function as it's only arguement and returns the unsubscribe function below that will unsubscribe to function originally passed to onAuthStateChanged
     return unsubscribe
   }, [])
 
+  // force refresh the token every 10 minutes
+  useEffect(() => {
+    const handle = setInterval(async () => {
+      console.log('refreshing!', new Date())
+      const user = auth.currentUser
+      if (user) await user.getIdToken(true)
+    }, 10 * 60 * 1000)
+
+    // clean up setInterval
+    return () => clearInterval(handle)
+  }, [])
+
```

Comment:
auth token from firebase expires after an hour, and don't get automatically refreshed, this refreshes the token every 10 minutes to ensure that we always have good token. 
source: https://colinhacks.com/essays/nextjs-firebase-authentication